components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2417,6 +2417,23 @@ public static String getUserNameProvisioningUIUrl() {
         return userNamePrvisioningUrl;
     }
 
+    /**
+     * To check OpenJDK Nashorn is availability.
+     *
+     * @return AdaptiveAuthentication Enabled or not.
+     */
+    public static boolean isAdaptiveAuthenticationAvailable() {
+
+        boolean isAdaptiveAuthenticationAvailable = false;
+        try {
+            Class.forName("org.openjdk.nashorn.api.scripting.ScriptObjectMirror");
+            isAdaptiveAuthenticationAvailable = true;
+            return isAdaptiveAuthenticationAvailable;
+        } catch (ClassNotFoundException e) {
+            return isAdaptiveAuthenticationAvailable;
+        }
+    }
```

Comment:
tring to do Class.forName each time is expensive, let's do this during the internal component activate logic and store the result in Dataholder and expose the value through this util.

Reply:
Fixed